plugin/src/main/java/com/gtnewhorizons/retrofuturagradle/modutils/DependencyDeobfuscationTransform.java

```diff
@@ -126,65 +112,112 @@ private void runTransform(TransformOutputs outputs) throws IOException {
-        }
 
-        Files.move(outFileTemp, outFile);
-    }
+                try (final OutputStream os = FileUtils.openOutputStream(outFileTemp, false);
+                        final BufferedOutputStream bos = new BufferedOutputStream(os);
+                        final JarOutputStream jos = mf != null ? new JarOutputStream(bos, mf)
+                                : new JarOutputStream(bos)) {
+                    final Enumeration<JarEntry> iter = ijar.entries();
+                    while (iter.hasMoreElements()) {
+                        final JarEntry entry = iter.nextElement();
+                        if (StringUtils.endsWith(entry.getName(), "META-INF/MANIFEST.MF")
+                                || StringUtils.endsWithIgnoreCase(entry.getName(), ".dsa")
+                                || StringUtils.endsWithIgnoreCase(entry.getName(), ".rsa")
+                                || StringUtils.endsWithIgnoreCase(entry.getName(), ".sf")
+                                || StringUtils.containsIgnoreCase(entry.getName(), "meta-inf/sig-")) {
+                            continue;
+                        }
+                        try (final InputStream is = ijar.getInputStream(entry);
+                                final BufferedInputStream bis = new BufferedInputStream(is)) {
+                            final String depName = containedDeps.get(entry.getName());
+                            if (depName != null) {
+                                File tempFile = File.createTempFile(depName, "-rfg_tmp_copy.jar", outputDir);
+                                tempFile.deleteOnExit();
+                                try (final OutputStream tos = FileUtils.openOutputStream(tempFile);
+                                        final BufferedOutputStream btos = new BufferedOutputStream(tos)) {
+                                    IOUtils.copy(bis, btos);
+                                }
+                                deobfQueue.add(
+                                        new DeobfJob(depName, tempFile, job.sourceName + "!" + entry.getName(), true));
+                                continue;
+                            }
 
-    private static JarOutputStream makeTransformerJarOutputStream(JarInputStream jis, OutputStream os)
-            throws IOException {
-        if (jis.getManifest() != null) {
-            final Manifest mf = jis.getManifest();
-            transformManifest(mf);
-            return new JarOutputStream(os, mf);
-        } else {
-            return new JarOutputStream(os);
+                            jos.putNextEntry(new JarEntry(entry.getName()));
+                            if (StringUtils.endsWithIgnoreCase(entry.getName(), ".class")) {
+                                byte[] data = IOUtils.toByteArray(bis);
+                                IOUtils.write(Utilities.simpleRemapClass(data, combined), jos);
+                            } else {
+                                IOUtils.copy(bis, jos);
```

Comment:
Wouldn't it be better to remap the embedded jars in-place instead of generating multiple output artifacts for one input artifact? This could be done with some recursion, and would maintain the input structure better.